src/portfft/committed_descriptor_impl.hpp

```diff
@@ -1058,20 +990,12 @@ class committed_descriptor_impl {
     std::size_t outer_size = total_size / params.lengths.back();
     std::size_t input_stride_0 = input_strides.back();
     std::size_t output_stride_0 = output_strides.back();
-    // distances are currently used just in the first dimension - these changes are meant for that one
-    // TODO fix this to support non-default layouts
-    if (input_stride_0 < input_distance) {  // for example: batch interleaved input
-      input_distance = params.lengths.back();
-    }
-    if (output_stride_0 < output_distance) {  // for example: batch interleaved output
-      output_distance = params.lengths.back();
-    }
 
     PORTFFT_LOG_TRACE("Dispatching the kernel for the last dimension");
     sycl::event previous_event =
         dispatch_kernel_1d(in, out, in_imag, out_imag, dependencies, params.number_of_transforms * outer_size,
-                           input_stride_0, output_stride_0, input_distance, output_distance, input_offset,
-                           output_offset, dimensions.back(), compute_direction);
+                           input_stride_0, output_stride_0, input_distance / outer_size, output_distance / outer_size,
```

Comment:
I've tried to clarify that here. https://github.com/codeplaysoftware/portFFT/pull/136/commits/3f12710323eb6a6a52d207a85da97d6af35d0467
The distance is the final dimension because the multidim transforms are always packed.

Reply:
I don't get how this works, the distance should be set to the total FFT size like we do here https://github.com/codeplaysoftware/portFFT/pull/136/files#diff-809aaf559d4b027a8bb3b0af31c4c9253a65cd4704510475f46acff32bc31478R142. This represents the number elements to move to get to the next batch. Does it have a different meaning for multi-dim FFTs that I forgot?

Reply:
In this case we are submitting only the contiguous dimension of the multi-dim fft, and since the only layout is packed, we have a `params.number_of_transforms * outer_size` transforms of length `params.lengths.back()`.
If you look into `dispatch_kernel_1d` and then `dispatch_kernel_1d_helper` you'll see that the distance is only being used to decide which layout to dispatch to, so this can be removed when all the strided work is done.

Reply:
`in/output_stride/distance_0` are just being used to decide the layout to dispatch to here, so I've made that clearer by directly passing the layout in: https://github.com/codeplaysoftware/portFFT/pull/136/commits/3672ad563da0c6940ab52fd4c0d248bedf403e78